client/question/services/code_preprocessors/PythonCodePreprocessorService.js

```diff
@@ -82,6 +84,12 @@ tie.factory('PythonCodePreprocessorService', [
      */
     var UPPER_BOUND_RATIO_IF_LINEAR = (LARGE_INPUT_SIZE / SMALL_INPUT_SIZE) * 3;
 
+    /**
+     * Used to separate the print outputs of different test cases, which are
+     * all combined into one output string after the code is run.
+     */
+    var separator = null;
```

Comment:
Better not to make this a global within this service.

Instead, in line 653, do something like `var separator = generateNewSeparator();` and then do generateOutputSeparatorCode(separator) when you need the code, and finally return the value of the local variable. There's no need for this to be global/persistent, it's a per-run thing.